CRYOSMSApp/src/CRYOSMSDriver.cpp

```diff
@@ -0,0 +1,119 @@
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <errno.h>
+#include <math.h>
+#include <exception>
+#include <iostream>
+#include <map>
+
+#include <epicsTypes.h>
+#include <epicsTime.h>
+#include <epicsThread.h>
+#include <epicsString.h>
+#include <epicsTimer.h>
+#include <epicsMutex.h>
+#include <epicsEvent.h>
+#include <errlog.h>
+#include <iocsh.h>
+#include <epicsExit.h>
+#include <dbCommon.h>
+#include <dbAccess.h>
+#include <boRecord.h>
+
+#include <asynPortDriver.h>
+#include <asynDriver.h>
+#include <asynStandardInterfaces.h>
+
+#include <epicsExport.h>
+
+#include "CRYOSMSDriver.h"
+
+
+static const char *driverName = "CRYOSMSDriver"; ///< Name of driver for use in message printing 
+
+CRYOSMSDriver::CRYOSMSDriver(const char *portName, std::string devPrefix)
+  : asynPortDriver(portName,
+	0, /* maxAddr */
+	NUM_SMS_PARAMS, /* num parameters */
+	asynInt32Mask | asynInt32ArrayMask | asynFloat64Mask | asynFloat64ArrayMask | asynOctetMask | asynDrvUserMask, /* Interface mask */
+	asynInt32Mask | asynInt32ArrayMask | asynFloat64Mask | asynFloat64ArrayMask | asynOctetMask,  /* Interrupt mask */
+	ASYN_CANBLOCK, /* asynFlags.  This driver can block but it is not multi-device */
+	1, /* Autoconnect */
+	0,
+	0)
+{
+	static const char *functionName = "asynPortDriver";
+	createParam(P_deviceNameString, asynParamOctet, &P_deviceName);
+	createParam(P_outputModeSetString, asynParamOctet, &P_outputModeSet);
+	this->devicePrefix = devPrefix;
+	
+}
+void CRYOSMSDriver::pollerTask()
+{
+}
+
+
+asynStatus CRYOSMSDriver::writeInt32(asynUser *pasynUser, epicsInt32 value)
+{
+	int function = pasynUser->reason;
+	if (function == P_outputModeSet) {
+		return putDb("HIDDEN:OUTPUTMODE:SP", &value);
+	}
+	else {
+		return asynSuccess;
+	}
+}
+
+
+
+asynStatus CRYOSMSDriver::putDb(std::string pvSuffix, const void *value) {
+	DBADDR addr;
+	std::string fullPV = this->devicePrefix + pvSuffix;
+	if (dbNameToAddr(fullPV.c_str(), &addr)) {
+		throw std::runtime_error("PV not found: " + fullPV);
```

Comment:
This exception will never be caught, likely crashing the IOC? Is that the expected behaviour?